src/vpn-worker.ts

```diff
@@ -102,67 +102,65 @@ const worker = async (instanceId: number, serviceId: number) => {
 		}
 	});
 
-	let draining = false;
```

Comment:
I think only allowing one draining attempt makes sense, but you could also do it via
```ts
const drainConnections = _.once(async () => {
```
if you want to simplify it (it won't have any logging for it though)